modules/collect/src/main/java/com/opengamma/strata/collect/MapStream.java

```diff
@@ -505,11 +519,40 @@ private MapStream(Stream<Map.Entry<K, V>> underlying) {
     return underlying.collect(Guavate.toImmutableMap(e -> e.getKey(), e -> e.getValue()));
   }
 
+  /**
+   * Concatenates the other map's entries to the underlying stream.
+   *
+   * @param otherMap the other map from which to get the entries to append
+   * @return the stream of entries including those newly concatenated
+   */
+  public MapStream<K, V> concat(Map<K, V> otherMap) {
```

Comment:
Why not have a `static` method similar to `Stream.concat` that takes `Map<? extends K, ? extends V> ...maps` and an overload that takes streams i.e `Stream<Map.Entry<? extends K, ? extends V>> ...mapStreams`?

I can't imagine a situation where one would call the instance method over the static method, especially since streams are single use so you wouldn't want to be passing a MapStream around.

Reply:
The instance method is slightly more flexible (can create a `MapStream` from any of the factory methods then `concat`) and less verbose to use, but a static method is more consistent with `Streams`/`Stream`. I'd be okay with making it static if we think it's neater

Reply:
See [here](https://stackoverflow.com/a/49616091/38896) for why `Stream` has a static `concat` method. Reluctantly, I agree with that conclusion, so I think we should follow the JDK exactly here. A `concat(MapStream, MapStream) static method` with generics similar to the [JDK method](https://download.java.net/java/early_access/jdk11/docs/api/java.base/java/util/stream/Stream.html#concat(java.util.stream.Stream,java.util.stream.Stream)).

Lets avoid other variants, such as ones with `Map` or varargs at the moment - they can be added later if needed.

Reply:
I'm made some initial these changes in this vein, but still looking at passing `Map<Integer, Double>` to `Map<Number, Object>` for example -  the type signature of `Stream.concat` seems to be restrictive in this case

Reply:
I've now added that test. With the signature as it stands, I couldn't get round callers having to choose between:
`ImmutableMap<Object, ? extends Number> result =
        MapStream.concat(MapStream.of(map1), MapStream.of(map2)).toMap((a,b) -> a);`
or 
`ImmutableMap<Object, Number> result =
        MapStream.<Object, Number>concat(MapStream.of(map1), MapStream.of(map2)).toMap((a,b) -> a);`

If you have any ideas for avoiding either one of these I'm all ears